blocks/layout-grid/style.scss

```diff
@@ -186,16 +186,22 @@
  * Individual column alignment
  */
 .wp-block-jetpack-layout-grid-column {
+	display: flex;
+	height: 100%;
+
 	&.is-vertically-aligned-top {
 		align-self: flex-start;
+		align-items: flex-start;
```

Comment:
It seems this is the key change that allows both a full height column, so the background color can stretch to fill the vertical space, and apply the alignment inside. Probably the reduction in containers made align-self need replacing by an align-items instead.